pkg/catalog/routes.go

```diff
@@ -397,3 +418,75 @@ func listTrafficTargetPermutations(name string, srcServiceList []service.MeshSer
 
 	return trafficPolicies
 }
+
+func (mc *MeshCatalog) getWeightedBackendsForService(svc service.MeshService) []service.WeightedService {
+	weightedSvcs := []service.WeightedService{}
+
+	// Retrieve the weighted clusters from traffic split
+	// Assumes no nested splits for now
+	// Assumes service would only be root once (not sure if this is validated)
+	servicesList := mc.meshSpec.ListTrafficSplitServices()
+	for _, activeService := range servicesList {
+		if activeService.RootService == svc.Name {
+			weightedSvcs = append(weightedSvcs, activeService)
+		}
+	}
```

Comment:
Correct me if wrong, this patch does not address https://github.com/openservicemesh/osm/issues/1798 right? (just making sure here)